### PR TITLE
[issue #22] Create binding for TH*Vector_* functions

### PR DESCRIPTION
**Pull request by motus**
This PR addresses issue #22

### CODE REVIEW

File: TorchSharp/Raw/Vector.tt [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+using System.Runtime.InteropServices;
+
+namespace TorchSharp.Raw {
+
+    internal static class Vector {
+
+<#
+foreach (var type in TorchTypeDef.Types) {
+#>
+        // Fill n elements of vector x with the value c.
```
Comment (interesaaat) [inline TorchSharp/Raw/Vector.tt]: What do you mean with fill n elements? Are these the first n elements, or skip the first n elements? (I see that n is ptrdiff)

File: TorchSharp/Raw/Vector.tt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        //
+        // Corresponds to the following TH declaration:
+        //
+        // TH_API void THVector_(fill)(scalar_t *x, const scalar_t c, const ptrdiff_t n);
+        [DllImport ("caffe2")]
+        internal extern static void TH<#=type.Name#>Vector_fill(
+            <#=type.Ptr#> x,
```
Comment (interesaaat) [inline TorchSharp/Raw/Vector.tt]: Can we have the parameters in one line? I see that this is what you are doing for the other functions.

File: TorchSharp/Raw/Vector.tt (hunk 1 above)
Comment (motus) [reply TorchSharp/Raw/Vector.tt]: `ptrdiff_t` is used like `size_t` here and in other functions, and `x` can be an offset into the vector. That is, the function will assign `c` to `n` values starting from pointer `x`. I will update the comment to clarify.

Comment (interesaaat) [reply TorchSharp/Raw/Vector.tt]: ah! I see. For some reason I was assuming that `ptrdiff_t` was the index in `x`.

**Comment by interesaaat** [comment]
One question, instead of having Raw with the tt files for BLAS, etc. should we be consistent with the project and having tt files and cs file named something like THBLAS and leaving them into the main directory?

**Comment by motus** [comment]
@interesaaat I actually don't like cluttering the main directory with files related to the low-level API. I would rather move the `extern TH*Tensor_*` functions into their own file and namespace in `Raw/`. I also think that having `TH*` prefix for high-level classes and modules is redundant - it does not not carry any additional information.  I am OK with keeping `TH*` prefix for low-level functions, though, so that the C# names match their C counterparts exactly.

**Comment by interesaaat** [comment]
Sounds good to me. Should we fill an issue to re-organize the project structure?

**Pull request merged**